app/models/concerns/user_omniauth_facebook_concern.rb

```diff
@@ -0,0 +1,17 @@
+module UserOmniauthFacebookConcern
+  extend ActiveSupport::Concern
+
+  module ClassMethods
+    # A convenience method that receives both parameters and session to initialize a user.
+    # This is a override of `Devise::Models::Registerable::ClassMethods#new_with_session`
+    def new_with_session(params, session)
+      super.tap do |user|
+        facebook_data = session['devise.facebook_data']
+        if facebook_data && (info = facebook_data['info'])
+          user.assign_attributes(name: info['name'], email: info['email'])
```

Comment:
wait, it is tested. okay, sure.

I think this should not matter if it is assigned after super, since it is a `new` and not a `create`?